### ISSUE TITLE
Gatsby route transition accessibility text rendered to page when using react-toast-notifications

### ISSUE THREAD

**Issue by kellenmace** [gatsbyjs/gatsby#21621]
## Description

When using the `react-toast-notifications` npm package and running a Gatsby production build, accessibility text indicating which page the user has navigated to gets rendered the page after each route transition. 

Here is a screenshot showing the issue:
https://cloudup.com/c4vh5mYHoSX

### Steps to reproduce

1. Create a new local site: `gatsby new toasts-test`
2. Install the npm package: `npm i react-toast-notifications`
3. Paste this into `gatsby-ssr.js` and `gatsby-browser.js` to add the provider:

```
import React from "react"
import { ToastProvider } from "react-toast-notifications"
export const wrapRootElement = ({ element }) => (
  <ToastProvider>{element}</ToastProvider>
)
```

4. Run `gatsby build && gatsby serve`, then visit the site at http://localhost:9000/.
5. Click the links to navigate between routes and notice that the “Navigated to...” accessibility text gets rendered to the screen.

##### Notes
- The accessibility text is inside of a div with a class of `react-toast-notifications__container`, so it’s clear that it’s being rendered via the `react-toast-notifications` library, but I have no idea how or why that text is being passed from Gatsby/Reach Router to that toast package's code.
- The accessibility text does NOT get rendered when running a dev server using `gatsby develop`– only when running a production build with `gatsby build && gatsby serve`.

##### CodeSandbox Example
I created a CodeSandbox here, with the steps above already done:
https://codesandbox.io/s/gatsby-toast-test-rob3g

As mentioned above, the issue only happens on production builds though, so you'll need to either export that CodeSandbox as a zip and run a prod build locally, or deploy the CodeSandbox example to now.sh or a similar service to reproduce it.

### Expected result

On route transitions, "Navigated to..." accessibility text should not be rendered to the page.

### Actual result

On route transitions, "Navigated to..." accessibility text is being rendered to the page.

### Environment

```
System:
    OS: macOS 10.15.3
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.13.1 - /usr/local/bin/node
    npm: 6.12.1 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 80.0.3987.116
    Firefox: 72.0.2
    Safari: 13.0.5
  npmPackages:
    gatsby: ^2.19.7 => 2.19.7
    gatsby-image: ^2.2.39 => 2.2.39
    gatsby-plugin-manifest: ^2.2.39 => 2.2.39
    gatsby-plugin-offline: ^3.0.32 => 3.0.32
    gatsby-plugin-react-helmet: ^3.1.21 => 3.1.21
    gatsby-plugin-sharp: ^2.4.3 => 2.4.3
    gatsby-source-filesystem: ^2.1.46 => 2.1.46
    gatsby-transformer-sharp: ^2.3.13 => 2.3.13
  npmGlobalPackages:
    gatsby-cli: 2.8.27
```

**Comment by pieh** [comment]
Ok, that probably shouldn't happen and should be considered a bug. But first let's involve @madalynrose on this

**Comment by pieh** [comment]
Verified the issue and have proposal of a fix: https://github.com/gatsbyjs/gatsby/pull/21625

**Comment by pieh** [comment]
After checking with @blainekasten we thing that the `react-toast-notifications` is doing few things wrong. Gatsby does as well to be fair (and we will need to make some changes to at least make Gatsby part behave correctly - there are some details in the PR I linked above).

Parts about `react-toast-notifications`:
 - It creates portal using `document.body` as a target:
https://github.com/jossmac/react-toast-notifications/blob/ef4f87413dfff55adece2464c667019d668adcca/src/ToastProvider.js#L173
which isn't good idea - https://stackoverflow.com/questions/49504546/is-it-safe-to-use-reactdom-createportal-with-document-body
 - it behaves differently in SSR and in Browser - https://github.com/jossmac/react-toast-notifications/blob/ef4f87413dfff55adece2464c667019d668adcca/src/ToastProvider.js#L179-L225 - in SSR it creates dom element in place (with the comment `// keep ReactDOM.hydrate happy`) and in browser it creates portal that will be mounted in different DOM element

I think that you can workaround this by actually removing `< ToastProvider/>` from `gatsby-ssr` (if that's all you have in that file, you could remove the file completely). This is not great fix overall, but at least should not conflict with gatsby announcer anymore.

**Comment by kellenmace** [comment]
@pieh Great! Thanks for looking into it and proposing a fix. Much appreciated.

In terms of the workaround you mentioned –
Removing `<ToastProvider/>` from `gatsby-ssr.js` results in a `The "useToasts" hook must be called from a descendent of the 'ToastProvider".` error at buildtime (`useToasts()` is the hook the library provides for interacting with toast notifications).

For the moment, I have a super hacky fix in place for my project: a useEffect callback that runs on each route transition and searches the DOM for a div with the class of `react-toast-notifications__container` whose `innerText` starts with `Navigated to` and sets it to `display="none"` 😬.

Looking forward to your proper fix getting into Gatsby core! Thanks again. 👍

**Comment by barbalex** [comment]
I can see the exact same issue but I am not using react-toast-notifications.

You can test it by:
- open https://evab.ch/events
- click "Actors" in the top navbar
- see "Navigated to Actors" under the content

I have absolutely no idea why this would happen.

I _did_ notice that it does _not_ happen when you do this:
- open https://evab.ch
- click "Actors" in the top navbar
- see _no_ "Navigated to Actors" under the content
- click any navbar link and it still does not appear

So it seems that the issue in my case is provoked when the page is initially opened not on the root.

So maybe it has something to do with the way I use @reach-router, see:
- https://github.com/barbalex/bb2/blob/master/src/App.js
- https://github.com/barbalex/bb2/blob/master/src/pages/actors.js

**Comment by sidharthachatterjee** [comment]
Fix published in `gatsby@2.19.43` courtesy of @pieh and @blainekasten

**Comment by barbalex** [comment]
Yep, that solved my issue. Thanks!
